test/Microsoft.VisualStudio.Threading.Tests/JoinableTaskContextTests.cs

```diff
@@ -45,6 +45,55 @@ public void IsWithinJoinableTask()
         });
     }
 
+    [Fact]
+    public void IsMainThreadBlockedByAnyone()
+    {
+        Assert.False(this.Context.IsMainThreadBlockedByAnyone);
+        AsyncManualResetEvent mainThreadBlockerEvent = new AsyncManualResetEvent(false);
+        AsyncManualResetEvent backgroundThreadMonitorEvent = new AsyncManualResetEvent(false);
+
+        // Start task to monitor IsMainThreadBlockedByAnyone
+        Task.Run(async () =>
+        {
+            await TaskScheduler.Default.SwitchTo(alwaysYield: true);
+            await mainThreadBlockerEvent.WaitAsync();
+
+            bool shouldWaitForMainThreadBlock = true;
+            while (shouldWaitForMainThreadBlock)
+            {
+                try
+                {
+                    Assert.True(this.Context.IsMainThreadBlockedByAnyone);
+                    shouldWaitForMainThreadBlock = false;
+                }
+                catch
+                {
+                    // Give the main thread time to enter a blocking state
+                    Thread.Sleep(50);
+                }
+            }
```

Comment:
This loops forever until the desired condition is met, throwing exceptions as a normal flow. It uses a local to escape a loop where `break` would suffice. But why not put the condition in the loop condition itself? Finally, why block a thread while waiting when you can do it asynchronously? We can rewrite like this:

```suggestion
            while (!this.Context.IsMainThreadBlockedByAnyone)
            {
                // Give the main thread time to enter a blocking state, if the test hasn't already timed out.
                this.TimeoutToken.ThrowIfCancellationRequested();
                await Task.Delay(AsyncDelay);
            }
```